adapter/service/outlvlswitch/service.go

```diff
@@ -0,0 +1,203 @@
+package outlvlswitch
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/futurehomeno/fimpgo"
+	"github.com/futurehomeno/fimpgo/fimptype"
+
+	"github.com/futurehomeno/cliffhanger/adapter"
+	"github.com/futurehomeno/cliffhanger/adapter/cache"
+)
+
+// Constants defining important properties specific for the service.
+const (
+	MaxLvl        = "max_lvl"
+	MinLvl        = "min_lvl"
+	SwitchType    = "sw_type" // "on_off" or "up_down"
+	TypeOnAndOff  = "on_off"
+	TypeUpAndDown = "up_down"
+)
+
+// DefaultReportingStrategy is the default reporting strategy used by the service for periodic reports.
+var DefaultReportingStrategy = cache.ReportOnChangeOnly()
+
+// Controller is an interface representing an actual device.
+// In a polling scenario implementation might require some safeguards against excessive polling.
+type Controller interface {
+	// LevelReport returns a current level value.
+	LevelSwitchLevelReport() (int64, error)
+	// BinanryReport returns a current binary value.
+	LevelSwitchBinaryReport() (bool, error)
+	// SetLvlCtrl sets a level value.
+	SetLevelSwitchLevel(value int64) error
+	// SetLevelWithDurationCtrl sets a level value over a specified duration in seconds.
+	SetLevelSwitchLevelWithDuration(value int64, duration int64) error
```

Comment:
Very good move with a dedicated method with duration, however, I think we should change the duration type to time.Duration on the controller and make conversion from int in service.